extensions/redis-cache/runtime/src/main/java/io/quarkus/cache/redis/runtime/RedisCacheImpl.java

```diff
@@ -218,6 +218,34 @@ public Uni<V> apply(RedisConnection connection) {
         });
     }
 
+    @Override
+    public <K, V> V compute(K key, Class<V> clazz, BiFunction<K, V, V> command) {
+        byte[] encodedKey = marshaller.encode(computeActualKey(encodeKey(key)));
+
+        return withConnection(connection -> watch(connection, encodedKey)
+                .chain(new GetFromConnectionSupplier<>(connection, clazz, encodedKey, marshaller))
+                .chain((value) -> connection.send(Request.cmd(Command.MULTI)).replaceWith(value))
+                .chain(previousValue -> Uni.createFrom().item(() -> command.apply(key, previousValue))
+                        .runSubscriptionOn(MutinyHelper.blockingExecutor(vertx.getDelegate())))
```

Comment:
Why do we have to do this? Is it because the method is blocking?

Reply:
well, can we assume that the provided method will always be non blocking ?